src/api/java/com/minecolonies/api/util/FireworkUtils.java

```diff
@@ -0,0 +1,87 @@
+package com.minecolonies.api.util;
+
+import com.minecolonies.api.colony.IColony;
+import net.minecraft.entity.item.EntityFireworkRocket;
+import net.minecraft.item.ItemDye;
+import net.minecraft.item.ItemFirework;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.math.AxisAlignedBB;
+
+import java.util.Random;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.*;
+
+/**
+ * Utility class for summoning in fireworks.
+ */
+
+public final class FireworkUtils
+{
+    /**
+     * Private constructor to hide the public one
+     */
+    private FireworkUtils() {
+
+    }
+
+    /**
+     *
+     *
+     * @param realaabb AABB of the building
+     * @param colony which colony to .getWorld() from
+     * @param newLevel how many fireworks to spawn in each corner
+     */
+    public static void spawnFireworksAtBuildingCorners(final AxisAlignedBB realaabb, final IColony colony, final int newLevel)
```

Comment:
Change the IColony to a World.
That way we can reuse it for all kinds of things.

Reply:
Also rename newLevel to the actuall usage: explosionLevel

Reply:
And rename the method to something more generic like: `spawnFireworksAtAABBCorners`